sql-parser/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/impl/OracleDMLStatementSQLVisitor.java

```diff
@@ -512,6 +515,31 @@ public ASTNode visitQueryBlock(final QueryBlockContext ctx) {
         if (null != ctx.modelClause()) {
             result.setModelSegment((ModelSegment) visit(ctx.modelClause()));
         }
+        Map<String, OracleSelectStatement> rowNumSelect = getRowNumSelect(result);
+        if (rowNumSelect.size() > 0) {
+            result.setRowNumSelect(rowNumSelect);
```

Comment:
Can we move this logic into StatementContext for processing? The SQLVisitor should be processed according to SQL semantics, and some processing for kernel functions should be placed in StatementContext.

Reply:
great! I had tried.